pkg/shell/base_index.js

```diff
@@ -481,6 +478,20 @@ function Router(index) {
 
             /* Add the child's group to all open channel messages */
             } else if (control.command == "open") {
+                // Prevent frames from malicious remote hosts from tampering with other machines
```

Comment:
This check is necessary, but I am not sure whether it is sufficient. What happens when code in an iframe tries to open its own socket to cockpit-ws? That also needs to fail, and we should have a test for that, no?

Reply:
It's *very much not* sufficient from a security POV. Frames can open a direct socket, and they can also traverse the DOM to run commands in a different frame. See https://issues.redhat.com/browse/COCKPIT-870 .

This PR is just closing one of these holes, and that's the one which we formerly considered API (the other two are "just" unintended security holes, and we don't need to be careful about supporting them). I.e. this is the bit which prevents "oopses" and discourages other projects from doing potentially dangerous and unintended things.

I haven't looked into whether there are legitimate use cases for a frame to directly open its own socket. Some of our pages (cockpit-files, playground speed test, GTK broadway example) open a raw channel with `"/cockpit/channel/" + cockpit.transport.csrf_token + '?' + query`, and the speed test even directly opens a new web socket:
```
websocket = new window.WebSocket(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token) 
```

All of these need to be reviewed, too. Note: all of this is a huge pile of work, we just need to start somewhere.

Reply:
I.e. the eventual goal is that frames won't have access to the login cookie and csrf_token any more. Then they also can't open direct web sockets any more.

Reply:
Ok, got it.